### PR TITLE
Frontend uncommitted changes

### PR DESCRIPTION
**Pull request by salilgupta1**

### CODE REVIEW

File: civis_jupyter_notebooks/assets/extensions/uncommitted_changes.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  var uncommitted_changes = function(elem) {
+    var settings = {
+      url: '/git/uncommitted_changes',
+      type: 'GET',
+      contentType: 'applicaton/json',
+      success: function(data) {
+        if (data.hasOwnProperty('dirty') && data.dirty) {
+          elem.css('display', 'inline-block');
+        } else {
+          elem.css('display', 'none');
```
Comment (mattyb) [inline civis_jupyter_notebooks/assets/extensions/uncommitted_changes.js]: Hrm, I think we should also have an indicator when the changes are committed. Check with Annie?

Comment (salilgupta1) [reply civis_jupyter_notebooks/assets/extensions/uncommitted_changes.js]: i'll check with her. but how would we distinguish between a user recently committing something and there being nothing for them to commit. We'd have to plug into the shell channel to capture any `git commit` commands being written.

**Comment by mattyb** [comment]
It's going to lag either way, right?

**Comment by salilgupta1** [comment]
screen shot 2017-12-12 at 9 44 34 am

**Comment by salilgupta1** [comment]
screen shot 2017-12-12 at 9 45 23 am

File: civis_jupyter_notebooks/assets/custom.css [hunk 2]
```diff
@@ -56,6 +56,15 @@ a.notebook-back-link span {
   padding: 2px;
 }
 
+#uncommitted_changes {
+  border: 1px solid red;
+  color: red;
```
Comment (mattyb) [inline civis_jupyter_notebooks/assets/custom.css]: should this be `red` or match `civisRed`? Same for `green`

**Pull request merged**